artiq/coredevice/i2c.py

```diff
@@ -33,6 +33,11 @@ def i2c_read(busno: TInt32, ack: TBool) -> TInt32:
     raise NotImplementedError("syscall not simulated")
 
 
+@syscall(flags={"nounwind", "nowrite"})
+def i2c_pca954x_select(busno: TInt32, address: TInt32, channel: TInt32, clear: TBool) -> TNone:
```

Comment:
Probably easier to do one-hot decoding in Rust than add a parameter. It's also backward-compatible.